Java/core/src/main/java/com/amazon/randomcutforest/RandomCutForest.java

```diff
@@ -290,6 +572,79 @@ public int getThreadPoolSize() {
         return threadPoolSize;
     }
 
+    // TODO change in next PR with the mappers and tests
+    public IStateCoordinator<?, ?> getUpdateCoordinator() {
+        return stateCoordinator;
+    }
+
+    public ComponentList<?, ?> getComponents() {
+        return components;
+    }
+
+    /**
+     * Returns a clean deep copy of the point. Current clean-ups include changing
+     * negative zero -0.0 to positive zero 0.0.
+     *
+     * @param point The original data point.
+     * @return a clean deep copy of the original point.
+     */
+    public static double[] cleanCopy(double[] point) {
```

Comment:
minor. this public method should not belong to this forest class.

Reply:
issue 239. However the cleanCopy is needed whenever the point is submitted to the forest (for update or scoring of any sort, we need to check for 0.0 and -0.0 in all of them). But it'll be good to broaden the discussion to other methods as well as protected classification instead of public.